lib/process-services-cloud/src/lib/form/services/content-cloud-node-selector.service.spec.ts

```diff
@@ -0,0 +1,191 @@
+    });
+
+    it('should be able to verify and return nodeId from given folderId', async () => {
+        getNodeSpy.and.returnValue(Promise.resolve(folderVariableValueResponseBody));
+        const relativePathNodeEntry = await service.verifyAndReturnNodeId('mock-folder-id');
+
+        expect(relativePathNodeEntry).toBe('mock-folder-id');
+        expect(getNodeSpy).toHaveBeenCalledWith('mock-folder-id', undefined);
+    });
+
+    it('should return defined alias nodeId if the relative path does not exists', async () => {
+        getNodeSpy.and.returnValues(Promise.reject('Relative does not exists'), Promise.resolve(aliasNodeResponseBody));
+        const aliasNodeId = await service.getNodeIdFromPath({ alias: 'mock-alias', path: 'mock-relativePath' });
+
+        expect(getNodeSpy).toHaveBeenCalledTimes(2);
+        expect(aliasNodeId).toEqual('mock-alias-node-id');
+    });
+
+    it('should fetch default nodeId if the folderId is not defined', async () => {
+        getNodeSpy.and.returnValue(Promise.resolve(aliasNodeResponseBody));
+        await service.verifyAndReturnNodeId('');
+
+        expect(getNodeSpy).toHaveBeenCalledWith('-my-', undefined);
+    });
+
+    it('should return default nodeId if the given folder is not exist', async () => {
+        getNodeSpy.and.returnValues(Promise.reject('Folder does not exists'), Promise.resolve(aliasNodeResponseBody));
+        const aliasNodeId = await service.verifyAndReturnNodeId('mock-folder-id');
+
+        expect(getNodeSpy).toHaveBeenCalledTimes(2);
+        expect(aliasNodeId).toEqual('mock-alias-node-id');
+    });
+
+    it('should be able to fetch nodeId from given alias', async () => {
+        getNodeSpy.and.returnValue(Promise.resolve(aliasNodeResponseBody));
+        const aliasNodeId = await service.getNodeIdFromPath({ alias: 'mock-alias', path: undefined });
+
+        expect(getNodeSpy).toHaveBeenCalledWith('mock-alias', undefined);
+        expect(aliasNodeId).toEqual('mock-alias-node-id');
+    });
+
+    it('should be able to open the content node select panel dialog', () => {
+        service.openUploadFileDialog('nodeId', 'single', true, true);
+
+        expect(openDialogSpy).toHaveBeenCalled();
+    });
+
+    it('should show a warning notification if the relative path is invalid/deleted', async () => {
+        getNodeSpy.and.returnValue(Promise.reject('Relative path does not exists'));
+
+        try {
+            await service.getNodeIdFromPath({ alias: 'mock-alias', path: 'mock-relativePath' });
+            fail('An error should have been thrown');
+        } catch (error) {
+            expect(error).toEqual('Relative path does not exists');
+        }
+
+        service.openUploadFileDialog('nodeId', 'single', true, true).subscribe(() => {
+            expect(openDialogSpy).toHaveBeenCalled();
+            expect(service.sourceNodeNotFound).toBe(true);
+            expect(showWarningSpy).toHaveBeenCalledWith('success');
```

Comment:
This is an always green test, if I remove the code that dispatches the notification it won't fail

Reply:
updated